### DISCUSSION TITLE
What threadpool graphql uses?

### DISCUSSION THREAD

**Discussion by hexwit** [graphql-java/graphql-java#2414] [Q&A] [answered]
Does graphql share the same thread pool as spring framework if both integrated in the same application? 
Is it possible to configure what thread pool to use for graphql? 

The main goal is to be able to use thread local defined on the controller level in the data fetcher.

Thanks.

**Comment by bbakerman** [comment, marked as answer]
graphql-java the library uses NO threadpool - its completely up to your implementation and how your DataFetchers dispatch results

Some libraries that wrap graphql-java may introduce threading but I cant comment on them.

See https://www.graphql-java.com/blog/threads/

**Reply by hexwit** [reply]
Thanks!
